lib/chef-dk/command/generator_commands/chef_exts/customized_doc_formatter.rb

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright:: Copyright (c) 2016 Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/formatters/doc'
+
+module ChefDK
+
+  # Subclass of Chef's standard 'doc' formatter that mutes messages that occur
+  # prior to convergence. This gives us cleaner output in general, but is
+  # especially noticeable when the standard formatter is disabled by the
+  # generator cookbook.
+  class CustomizedDocFormatter < Chef::Formatters::Doc
```

Comment:
It's not really silenced, you would still see output like 

```
Recipe: code_generator::cookbook
  * directory[/Users/ddeleo/sandbox/uw-19-sandbox/foo] action create
    - create new directory /Users/ddeleo/sandbox/uw-19-sandbox/foo
  * template[/Users/ddeleo/sandbox/uw-19-sandbox/foo/metadata.rb] action create_if_missing
    - create new file /Users/ddeleo/sandbox/uw-19-sandbox/foo/metadata.rb
    - update content in file /Users/ddeleo/sandbox/uw-19-sandbox/foo/metadata.rb from none to 8aded9
    (diff output suppressed by config)

(ETC)
```

Does `QuieterDocFormatter` sound ok?